File: functions/src/models/PhotoDetails.ts
/* eslint-disable */



export class PhotoDetails {

    private filePath!:string;
    private photoDownloadUrl!:string;
    private priority!:number;
    private createdAtUTC!:number;


    constructor() {
    }

    public static makeInstanceFromSnapshot(snapshotNeedsParsing:any): PhotoDetails {

        const photoDetails:PhotoDetails = new PhotoDetails();

        let parsedSnapshot;
        try{
            parsedSnapshot = JSON.parse(snapshotNeedsParsing);
        }catch (Error){
            parsedSnapshot = snapshotNeedsParsing;
        }


        photoDetails.setFilePath(parsedSnapshot.filePath);
        photoDetails.setPhotoDownloadUrl(parsedSnapshot.photoDownloadUrl);
        photoDetails.setPriority(parsedSnapshot.priority);
        photoDetails.setCreatedAtUTC(parsedSnapshot.createdAtUTC);

        return photoDetails;
    }

    static makeInstance(filePath:string, priority:number, photoDownloadUrl:string) {

        const photoDetails:PhotoDetails = new PhotoDetails();


        photoDetails.setFilePath(filePath);
        photoDetails.setPhotoDownloadUrl(photoDownloadUrl);
        photoDetails.setPriority(priority);
        photoDetails.setCreatedAtUTC(Date.now());

        return photoDetails;
    }

    toObject() {//this is the output that goes to firestore so becareful with the changes in this class

        return {

            filePath: this.filePath,
            photoDownloadUrl: this.photoDownloadUrl,
            priority: this.priority,
            createdAtUTC: this.createdAtUTC,
        };
    }


    public getFilePath(): string {
        return this.filePath;
    }

    public setFilePath(value: string) {
        this.filePath = value;
    }

    public getPhotoDownloadUrl(): string {
        return this.photoDownloadUrl;
    }

    public setPhotoDownloadUrl(value: string) {
        this.photoDownloadUrl = value;
    }

    public getPriority(): number {
        return this.priority;
    }

    public setPriority(value: number) {
        this.priority = value;
    }

    public getCreatedAtUTC(): number {
        return this.createdAtUTC;
    }

    public setCreatedAtUTC(value: number) {
        this.createdAtUTC = value;
    }


}

File: app/src/main/java/com/baikas/sporthub6/models/PhotoDetails.java
package com.baikas.sporthub6.models;

import com.baikas.sporthub6.interfaces.ObjectWithId;
import com.google.firebase.firestore.Exclude;

import java.io.Serializable;
import java.util.Map;

public class PhotoDetails implements ObjectWithId, Serializable {

    private String filePath;
    private String photoDownloadUrl;
    private long priority;
    private long createdAtUTC;

    public PhotoDetails() {
    }


    public PhotoDetails(String filePath, String photoDownloadUrl, long priority, long createdAtUTC) {
        this.filePath = filePath;
        this.photoDownloadUrl = photoDownloadUrl;
        this.priority = priority;
        this.createdAtUTC = createdAtUTC;
    }

    public PhotoDetails(Map<String,Object> map) {
        this.filePath = (String) map.get("filePath");
        this.photoDownloadUrl = (String) map.get("photoDownloadUrl");

        if (map.get("priority") instanceof Integer){
            this.priority = ((Integer) map.get("priority")).longValue();
        }else
            this.priority = (Long) map.get("priority");

        if (map.get("createdAtUTC") instanceof Integer){
            this.createdAtUTC = ((Integer) map.get("createdAtUTC")).longValue();
        }else
            this.createdAtUTC = (Long) map.get("createdAtUTC");

    }

    public PhotoDetails(PhotoDetails photoDetails) {
        this.filePath = photoDetails.getFilePath();
        this.photoDownloadUrl = photoDetails.getPhotoDownloadUrl();

        this.priority = photoDetails.getPriority();
        this.createdAtUTC = photoDetails.getCreatedAtUTC();
    }

    public String getPhotoDownloadUrl() {
        return photoDownloadUrl;
    }

    public void setPhotoDownloadUrl(String photoDownloadUrl) {
        this.photoDownloadUrl = photoDownloadUrl;
    }

    public long getPriority() {
        return priority;
    }

    public void setPriority(long priority) {
        this.priority = priority;
    }

    public String getFilePath() {
        return filePath;
    }

    public void setFilePath(String filePath) {
        this.filePath = filePath;
    }

    public long getCreatedAtUTC() {
        return createdAtUTC;
    }

    public void setCreatedAtUTC(long createdAtUTC) {
        this.createdAtUTC = createdAtUTC;
    }

    @Override
    @Exclude
    public String getId() {
        return photoDownloadUrl;
    }
}
